dace/codegen/targets/cuda.py

```diff
@@ -1238,6 +1238,13 @@ def _emit_copy(self, state_id: int, src_node: nodes.Node, src_storage: dtypes.St
                 dims = len(copy_shape)
 
                 funcname = 'dace::%sTo%s%dD' % (_get_storagename(src_storage), _get_storagename(dst_storage), dims)
+
+                # Check for GlobalToGlobal copies which are not well-defined
+                if (src_storage == dtypes.StorageType.GPU_Global and dst_storage == dtypes.StorageType.GPU_Global):
+                    raise NotImplementedError(
+                        "GPU global memory to global memory copies need to be more explicitly specified in the code. "
+                        "Consider using shared memory, different memory scopes, or explicit synchronization patterns.")
+
```

Comment:
Doesn't the not implemented error sound random?
Glb -> Glb being more explicitly specified in a code means that it needs to be copied within a map (e.g. calling copytomap as a transformation)? 
As if there is a program that absolutely needs glb to glb memory copy (or the user wants that), I think asking for different memory locations to be used is irrelevant. Different memory scopes wont help and we don't have explicit synchronization patterns in dace (also what will adding a syncthreads will do)?

Reply:
What about:

NotImplementedError(
                        "GPU global memory to global memory copies need to be more explicitly specified in the code. Considering moving the copy to a map using the CopyToMap transformation or manually. "
                      )